src/document.rs

```diff
@@ -276,3 +304,857 @@ impl Display for Created {
         write!(f, "{}", repr)
     }
 }
+
+impl From<&cargo_metadata::Package> for Package {
+    fn from(package: &cargo_metadata::Package) -> Self {
+        Package {
+            name: package.name.to_string(),
+            spdxid: format!("SPDXRef-{}-{}", package.name, package.version),
+            version_info: Some(package.version.to_string()),
+            package_file_name: None,
+            supplier: None,
+            originator: None,
+            download_location: NOASSERTION.to_string(),
+            files_analyzed: None,
+            package_verification_code: None,
+            checksums: None,
+            homepage: package.homepage.clone(),
+            source_info: None,
+            license_concluded: NOASSERTION.to_string(),
+            license_declared: NOASSERTION.to_string(),
+            copyright_text: NOASSERTION.to_string(),
+            description: None,
+            comment: None,
+            external_refs: if package.source.is_some() {
```

Comment:
Do we only want external_refs for non-local dependencies? Should we add them for local ones anyway?

If the author of a program written in Rust publishes the crate after they've built the binary, then having an external ref is useful to consumers of the binary, as they can check for RustSecs in the local crate itself.

(I think this is one of the reasons for gitBOM's existence :) )

Reply:
I don't think it hurts to include! 😄

Reply:
ok done